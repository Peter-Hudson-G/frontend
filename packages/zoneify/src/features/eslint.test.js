const { addDependency, addPackageJsonChange } = require('../install');
const { name, questions, run } = require('./eslint');

jest.mock('../install');

describe('features/eslint', () => {
  test('exports a name', () => {
    expect(name).toMatchSnapshot();
  });

  test('exports questions', () => {
    expect(questions).toMatchSnapshot();
  });

  test('exports run function', () => {
    expect(typeof run).toBe('function');
  });

  describe.each([
    ['no framework', '', '@zonedigital/eslint-config'],
    ['React', 'react', '@zonedigital/eslint-config/react'],
    ['Vue', 'vue', '@zonedigital/eslint-config/vue'],
  ])('runs with %s selected', (option, framework, configPath) => {
    beforeAll(() => {
      addDependency.mockReset();
      addPackageJsonChange.mockReset().mockImplementation(cb => cb);

      run({ framework });
    });

    test('adds a dependency', () => {
      expect(addDependency).toBeCalledWith({
        name: '@zonedigital/eslint-config',
        includePeerDependencies: true,
      });
    });

    test('adds a config change', () => {
      expect(
        addPackageJsonChange.mock.results[0].value({ name: 'test package' })
      ).toEqual({
        name: 'test package',
        eslintConfig: {
          extends: configPath,
        },
      });
    });
  });
});
