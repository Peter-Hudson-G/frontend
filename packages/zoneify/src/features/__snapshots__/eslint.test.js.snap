// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/eslint exports a name 1`] = `"ESLint"`;

exports[`features/eslint exports questions 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "name": "None",
        "value": "",
      },
      Object {
        "name": "React",
        "value": "react",
      },
      Object {
        "name": "Vue",
        "value": "vue",
      },
    ],
    "message": "What type of JS framework are you using?",
    "name": "framework",
    "type": "list",
  },
]
`;
